packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/ManifestFactory.kt

```diff
@@ -29,9 +30,23 @@ object ManifestFactory {
     @Throws(JSONException::class)
     fun getEmbeddedManifest(manifestJson: JSONObject, configuration: UpdatesConfiguration?): Manifest {
         return if (manifestJson.has("releaseId")) {
-            LegacyManifest.fromLegacyRawManifest(manifestJson as LegacyRawManifest, configuration!!)
+            LegacyManifest.fromLegacyRawManifest(LegacyRawManifest(manifestJson), configuration!!)
         } else {
-            BareManifest.fromManifestJson(manifestJson as BareRawManifest, configuration!!)
+            BareManifest.fromManifestJson(BareRawManifest(manifestJson), configuration!!)
+        }
+    }
+
+    fun getRawManifestFromJson(manifestJson: JSONObject): RawManifest {
+        return when {
+            manifestJson.has("releaseId") -> {
+                LegacyRawManifest(manifestJson)
+            }
+            manifestJson.has("updateMetadata") -> {
```

Comment:
this is the key to this whole PR. If we can't infer what type of manifest a manifest is from the manifest itself, this whole approach breaks down.